packages/cmf/src/cmfModule.js

```diff
@@ -1,21 +1,31 @@
 import merge from './cmfModule.merge';
 
-function find(options, buff = []) {
+async function getModule(module) {
+	const { init, ...syncModule } = module;
+	if (init) {
+		const asyncModule = await init();
+		return {
+			...syncModule,
+			...asyncModule,
+		};
+	}
+	return syncModule;
+}
+
+async function find(options, buff = []) {
 	if (options.modules) {
-		return options.modules.reduce((acc, current) => {
+		for await (const current of options.modules) {
 			if (!current.id) {
 				throw new Error('a cmf.module must have an id');
 			}
-			if (acc.some(({ id }) => current.id === id)) {
-				console.warn(
-					`cmf.bootstrap: 2 modules have the same id ${current.id}. This duplicated module will be skipped.`,
-				);
-			} else {
-				acc.push(current);
-				find(current, buff);
+
+			// prevent module duplication
+			if (!buff.some(({ id }) => current.id === id)) {
```

Comment:
I think we should let the console.warn, because module duplication is usually a mistake. If we don't warn the user, we don't offer a chance to fix it.